cabal-install/src/Distribution/Client/Types/SourcePackageDb.hs

```diff
@@ -21,3 +26,38 @@ data SourcePackageDb = SourcePackageDb
   deriving (Eq, Generic)
 
 instance Binary SourcePackageDb
+
+-- | Does a case-sensitive search by package name and a range of versions.
+--
+-- We get back any number of versions of the specified package name, all
+-- satisfying the version range constraint.
+--
+-- Additionally, `preferred-versions` (such as version deprecation) are
+-- honoured in this lookup, which is the only difference to
+-- 'PackageIndex.lookupDependency'
+lookupDependency :: SourcePackageDb -> PackageName -> VersionRange -> [UnresolvedSourcePackage]
```

Comment:
Is this the right location for this kind of function?

Reply:
I think so. The first used type is defined in this module and there are precedents of functions in such modules. Also, you didn't need to import any far away modules to make it work.